gb-apu/src/channel/duty.rs

```diff
@@ -0,0 +1,26 @@
+const DUTY_CYCLES: [[u8; 8]; 4] = [
+    [0, 0, 0, 0, 0, 0, 0, 1],
+    [1, 0, 0, 0, 0, 0, 0, 1],
+    [1, 0, 0, 0, 0, 1, 1, 1],
+    [0, 1, 1, 1, 1, 1, 1, 0],
+];
+
+#[derive(Default, Debug)]
+pub struct Duty {
+    pub pattern_index: u8,
+    step: u8,
+}
+
+impl Duty {
+    pub fn reset(&mut self) {
+        self.step = 0;
+    }
+    pub fn step(&mut self) {
+        self.step += 1;
+        self.step %= 8;
```

Comment:
```suggestion
        self.step = (self.step + 1) % 8;
```

maybe change the `8` to a constant that could be also used in defining `DUTY_CYCLES`